### ISSUE TITLE
[PATCH] add consumeErrors flag to DeferredList

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10368]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| spiv reported|
|-|-|
|Trac ID|trac#528|
|Type|enhancement|
|Created|2004-02-27 08:30:29Z|

Attachments:

* [consumeErrors.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/1d7/1d7deb7af392ce1dd0b07cc0e31a54ffbcbfe9fc/bc2d1099d0ca43031c953ab150a3225f43491997.patch) (1222 bytes) - added by spiv on 2004-02-27 08:30:29Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__528 528
type__enhancement enhancement
reporter__spiv spiv
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1077870629000000 1077870629000000
changetime__1472327697992280 1472327697992280
version__ 
owner__ 
cc__glyph cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
This is a quick patch to add a consumeErrors flag to DeferredList.  Judging by
recent discussion on the mailing list, this simple feature takes care of the
primary use-case of the now-deprecated addDeferred method.

Obviously the patch needs to grow some test cases and docs before it can be
checked in.

Discussion of this on the mailing list should continue in this issue.  

Any comments?
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
I had a use for this feature today while fixing a bug in FTPClient.  Seeing as
no-one has objected (or even commented ;) I've added some docs and tests, and
have checked this in.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

In changeset c712e86b89cb94b7ca7c7bd482cd6afa0be3a43a
```
#!CommitTicketReference repository="" revision="c712e86b89cb94b7ca7c7bd482cd6afa0be3a43a"
Merge pull request #528 from twisted/8789-rodrigc-strports

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8789

Make listen() and parse() visible symbols on Python 3
```
